### PR TITLE
{ARM} Bump azure-mgmt-resource SDK:  10.3.0 to 12.0.0,  v2020-06-01 to v2020-10-01

### PR DESCRIPTION
**Pull request by detienne20**
SDK upgraded to allow support for query string parameter for deployments (Model TemplateLink has a new parameter query_string)

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

**Comment by zhoxing-ms** [comment]
@detienne20 I see the description in [pypi.org](https://pypi.org/project/azure-mgmt-resource/11.0.0/) that there are some breaking changes in the Python SDK of version `11.0.0`
Could you please run the test of resource module again in live mode?

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 1]
```diff
@@ -2586,7 +2586,7 @@ class ManagedAppDefinitionScenarioTest(ScenarioTest):
     def test_managedappdef(self, resource_group):
 
         self.kwargs.update({
-            'upn': self.create_random_name('testuser', 15) + '@azuresdkteam.onmicrosoft.com',
+            'upn': self.create_random_name('testuser', 15) + '@detienne001gmail.onmicrosoft.com',
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2589]: Could you not change this email address to personal email address? Is this related to the issue of permissions?

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 2]
```diff
@@ -2651,7 +2651,7 @@ def test_managedappdef_inline(self, resource_group):
         curr_dir = os.path.dirname(os.path.realpath(__file__))
 
         self.kwargs.update({
-            'upn': self.create_random_name('testuser', 15) + '@azuresdkteam.onmicrosoft.com',
+            'upn': self.create_random_name('testuser', 15) + '@detienne001gmail.onmicrosoft.com',
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2654]: Same as above

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 3]
```diff
@@ -2719,7 +2719,7 @@ class ManagedAppScenarioTest(ScenarioTest):
     def test_managedapp(self, resource_group):
 
         self.kwargs.update({
-            'upn': self.create_random_name('testuser', 15) + '@azuresdkteam.onmicrosoft.com',
+            'upn': self.create_random_name('testuser', 15) + '@detienne001gmail.onmicrosoft.com',
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2722]: Same as above

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 4]
```diff
@@ -2971,9 +2971,9 @@ def test_deployment_group_cross_tenant(self, resource_group):
             'image': self.create_random_name('cli_crosstenantimage', 40),
             'version': '1.1.2',
             'captured': self.create_random_name('cli_crosstenantmanagedimage', 40),
-            'aux_sub': '1c638cf4-608f-4ee6-b680-c329e824c3a8',
+            'aux_sub': '6898adc8-5045-473d-a1bf-7012564f43cb',
             'rg': self.create_random_name('cli_test_cross_tenant_rg', 40),
-            'aux_tenant': '72f988bf-86f1-41af-91ab-2d7cd011db47'
+            'aux_tenant': '01a4073e-87c8-47cd-aafc-1439b4b5ea2c'
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2976]: Please do not change the `aux_tenant` and `aux_sub` of this test. If you don't have the permission to execute this test, I can rerecord yaml files instead of you

**Comment by zhoxing-ms** [comment]
What is `loa1.pdf`? Can we delete it?

File: src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_eventhub_commands_namespace_byok_test.py [hunk 5]
```diff
@@ -31,7 +31,7 @@ def test_eh_namespace_byok(self, resource_group):
             'tags2': {'tag2=value2'},
             'sku': 'Standard',
             'tier': 'Standard',
-            'clusterarmid': '/subscriptions/326100e2-f69d-4268-8503-075374f62b6e/resourceGroups/prod-by3-533-rg/providers/Microsoft.EventHub/clusters/PMTestCluster',
+            'clusterarmid': '/subscriptions/6898adc8-5045-473d-a1bf-7012564f43cb/resourceGroups/prod-by3-533-rg/providers/Microsoft.EventHub/clusters/PMTestCluster',
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_eventhub_commands_namespace_byok_test.py:34]: Why change this subscription?

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/management_group_level_parameters.json [hunk 6]
```diff
@@ -3,7 +3,7 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
     "nestedsubId": {
-      "value": "0b1f6471-1bf0-4dda-aec3-cb9272f09590"
+      "value": "a1bfa635-f2bf-42f1-86b5-848c674fc321az"
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/management_group_level_parameters.json:6]: Please do not change the `nestedsubId` of this test. If you don't have the permission to execute this test, I can rerecord yaml files instead of you

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/tenant_level_template.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
     },
     "nestedSubId": {
       "type": "string",
-      "defaultValue": "0b1f6471-1bf0-4dda-aec3-cb9272f09590"
+      "defaultValue": "6898adc8-5045-473d-a1bf-7012564f43cb"
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/tenant_level_template.json:11]: Please do not change the `nestedSubId` of this test. If you don't have the permission to execute this test, I can rerecord yaml files instead of you

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py [hunk 8]
```diff
@@ -455,7 +455,7 @@ def test_role_assignment_handle_conflicted_assignments(self, resource_group):
         with mock.patch('azure.cli.command_modules.role.custom._gen_guid', side_effect=self.create_guid):
             user = self.create_random_name('testuser', 15)
             self.kwargs.update({
-                'upn': user + '@azuresdkteam.onmicrosoft.com',
+                'upn': user + '@detienne001gmail.onmicrosoft.com',
                 'nsg': 'nsg1'
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py:459]: Same as above

File: src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_eventhub_commands_namespace_byok_test.py (hunk 5 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/eventhubs/tests/latest/test_eventhub_commands_namespace_byok_test.py:34]: I recorded this live and it changed to the subscription I used.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/management_group_level_parameters.json (hunk 6 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/management_group_level_parameters.json:6]: I do not have the permissions to execute this test, please do.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/tenant_level_template.json (hunk 7 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/tenant_level_template.json:11]: I do not have the permissions to execute this test, please do.

File: src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py (hunk 8 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/role/tests/latest/test_role.py:459]: I do not have the permissions to execute this test, please do.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 1 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2589]: Yes, I do not have the permissions, I had to use my personal account and change the domain to be able to record this test. If you can please record.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 2 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2654]: I do not have the permissions, I had to use my personal account and change the domain to be able to record this test. If you can please record.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 4 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2976]: Yes this is an issue of permissions.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 3 above)
Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2722]: I answered above.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:2589]: You can try to provide me with a remote virtual environment (such as VM) where I can log in to my test account, and then you can run the releated test in this environment with my account. This can avoid failures caused by permission problems.
Or I could run these tests for you and send the rerecord yaml file to you~

**Review by qwordy — APPROVED** [review]
No code change in VM part.

**Comment by detienne20** [comment]
> What is `loa1.pdf`? Can we delete it?

Yes I will delete it.

**Review by fengzhou-msft — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]
Approve  for ARM part

**Comment by msyyc** [comment]
No code change in network part

**Review by msyyc — APPROVED** [review]

**Review by v-Ajnava — APPROVED** [review]
LGTM

**Review by kairu-ms — APPROVED** [review]

**Review by Juliehzl — APPROVED** [review]

**Review by digimaun — APPROVED** [review]

**Review by aim-for-better — APPROVED** [review]
The impact for HDInsight is test files. I think it is ok. LGTM
